modules/core/src/system.cpp

```diff
@@ -1691,7 +1704,14 @@ class ThreadID {
         id(CV_XADD(&g_threadNum, 1))
     {
 #ifdef OPENCV_WITH_ITT
-        __itt_thread_set_name(cv::format("OpenCVThread-%03d", id).c_str());
+        std::string thread_name = cv::format("OpenCVThread-%03d", id);
+#if defined _WIN32 && defined _UNICODE
+        std::wstring ws_thread_name;
+        ws_thread_name.assign(thread_name.begin(), thread_name.end());
+        __itt_thread_set_name(ws_thread_name.c_str());
```

Comment:
According to PR's description this code should be disabled:

> -DCV_TRACE=OFF

Reply:
You are correct, I have removed this change